subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -235,12 +234,7 @@ public void execute(T plugin) {
                 // Test configuration extends main configuration
                 testSuite.getImplementationDependencies().extendsFrom(testedComponent.getImplementationDependencies());
                 ((Configuration)(testSuite.getDevelopmentBinary().getCompileModules())).getDependencies()
-                    .add(new DefaultSelfResolvingDependency((FileCollectionInternal)project.files(((DefaultSwiftBinary)testedComponent.getDevelopmentBinary()).getModuleFile().map(new Transformer<File, RegularFile>() {
-                        @Override
-                        public File transform(RegularFile regularFile) {
-                            return regularFile.getAsFile().getParentFile();
-                        }
-                    }))));
+                    .add(new DefaultSelfResolvingDependency((FileCollectionInternal)project.files(((DefaultSwiftBinary)testedComponent.getDevelopmentBinary()).getModuleFile())));
```

Comment:
I think we can replace the construction of a `DefaultSelfResolvingDependency` with a call to `project.getDependencies().create(project.files(testedComponent.getDevelopmentBinary().getModuleFile())`

Reply:
I will look into that.